apps/meteor/tests/end-to-end/api/livechat/contacts.ts

```diff
@@ -0,0 +1,277 @@
+import { faker } from '@faker-js/faker';
+import { expect } from 'chai';
+import { before, after, describe, it } from 'mocha';
+
+import { getCredentials, api, request, credentials } from '../../../data/api-data';
+import { createCustomField, deleteCustomField } from '../../../data/livechat/custom-fields';
+import { createAgent } from '../../../data/livechat/rooms';
+import { removePermissionFromAllRoles, restorePermissionToRoles, updatePermission, updateSetting } from '../../../data/permissions.helper';
+import { createUser } from '../../../data/users.helper';
+
+describe('LIVECHAT - contacts', () => {
+	before((done) => getCredentials(done));
+
+	before(async () => {
+		await updateSetting('Livechat_enabled', true);
+		await updatePermission('create-livechat-contact', ['admin']);
+	});
+
+	after(async () => {
+		await restorePermissionToRoles('create-livechat-contact');
+		await updateSetting('Livechat_enabled', false);
+	});
+
+	describe('[POST] omnichannel/contacts', () => {
+		it('should be able to create a new contact', async () => {
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+				});
+
+			expect(res.body).to.have.property('success', true);
+			expect(res.body).to.have.property('contactId');
+			expect(res.body.contactId).to.be.an('string');
+		});
+
+		it("should return an error if user doesn't have 'create-livechat-contact' permission", async () => {
+			await removePermissionFromAllRoles('create-livechat-contact');
+
+			const res = await request
+				.post(api('omnichannel/contacts'))
+				.set(credentials)
+				.send({
+					name: faker.person.fullName(),
+					emails: [faker.internet.email().toLowerCase()],
+					phones: [faker.phone.number()],
+				});
+
+			expect(res.body).to.have.property('success', false);
```

Comment:
does the permission get restored if the assertion fails??

Reply:
We restore the permission on the `after` function